### PR TITLE
Sample config: update LOCAL_ROOT_URL description for PROTOCOL = 'unix' case

### LINKED ISSUE #3741: Unix socket: cannot pull/push over SSH
- Gitea version (or commit ref): 1.4.0
- Git version: 2.16.3
- Go version: 1.10
- Operating system: ArchLinux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant

## Description

With the following configuration (actual domain redacted by example.com) in the `[server]` section:

```
PROTOCOL                   = unix
HTTP_ADDR                  = /run/nginx/gitea.sock
DOMAIN                     = git.example.com
ROOT_URL                   = https://git.example.com/
LOCAL_ROOT_URL = %(PROTOCOL)s://%(HTTP_ADDR)s:%(HTTP_PORT)s/
```

Trying to perform a git pull/push via SSH results in an error : 

```
UpdatePublicKey: Post unix:///run/nginx/gitea.sockapi/internal/ssh/1/update: unsupported protocol scheme "unix"
```

Looks like a regression over https://github.com/go-gitea/gitea/issues/2232, which has been fixed months ago. Adding some good old `Printf` traces in the code, I can see that `req.SetTransport` is called in `newInternalRequest`, but the `Dial` function that is set up there is never called ; the error is encountered before then. Changing `LOCAL_ROOT_URL` to `http://%(DOMAIN)s/` fixes the issue : both the `req.SetTransport` and `Dial` functions are called, and the request is performed on the unix socket (no log in the reverse proxy that’s listening on https://git.example.com/, and adding `deny all` on the reverse proxy do not prevent SSH pull/push), but that does not looks like to be the intended behavior.

### PR DESCRIPTION
**Pull request by Siosm**
Update the sample configuration description for the LOCAL_ROOT_URL
configuration variable to mention the special case required if PROTOCOL
is set to 'unix'.

Fixes #3741.

### CODE REVIEW

File: custom/conf/app.ini.sample [hunk 1]
```diff
@@ -122,6 +122,8 @@ UNIX_SOCKET_PERMISSION = 666
 ; Local (DMZ) URL for Gitea workers (such as SSH update) accessing web service.
 ; In most cases you do not need to change the default value.
 ; Alter it only if your SSH server node is not the same as HTTP node.
+; If you set PROTOCOL to 'unix', you must comment this variable or set it to
+; 'http://unix/'
 LOCAL_ROOT_URL = %(PROTOCOL)s://%(HTTP_ADDR)s:%(HTTP_PORT)s/
```
Comment (techknowlogick) [inline custom/conf/app.ini.sample]: Can you also update the docs page that references the setting: https://docs.gitea.io/en-us/config-cheat-sheet/

**Review by techknowlogick — CHANGES_REQUESTED** [review]

**Comment by Siosm** [comment]
Updated and rebased on master.

**Review by lafriks — APPROVED** [review]

File: custom/conf/app.ini.sample (hunk 1 above)
Comment (daviian) [reply custom/conf/app.ini.sample]: I suggest: "Don't set this variable if PROTOCOL is 'unix'"

**Comment by Siosm** [comment]
Re-updated with a slightly simpler formulation more in line with other doc comments.

**Review by techknowlogick — APPROVED** [review]

**Comment by Siosm** [comment]
Typo fix for 'unix' -> `unix` and re-updated my repo as something strange happened.

**Review by daviian — APPROVED** [review]

**Pull request merged**